### PR TITLE
VideoPress block: Adding new wpcom endpoint for updating VideoPress metadata.

### PR DESCRIPTION
**Pull request by roundhill**
#### Changes proposed in this Pull Request:
This PR adds a new `/wpcom/v2/videopress/meta` endpoint that we can utilize for future improvements of the VideoPress block to update metadata about the video.

~_Note: This currently won't work because of a bug on the wp.com api. Should be fixed soon._~ Fixed :)

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

 * Open a post in the block editor and open the browser console, and call this code for a VideoPress Id that the site has access to:
`wp.apiFetch( {
    path: '/wpcom/v2/videopress/meta',
    method: 'POST',
    data: { guid: {your-video-guid-goes-here}, rating: 'R-17' },
} ).then( res => {
    console.log( res );
} );`
 * If the request goes through, you should see `{code: "success", message: "Video meta updated successfully.", data: 200}` in the console and the age gate for your video when viewed.
 * Requests should fail with an object like `{code: "unknown_media", message: "The specified video was not found.", data: 404}` if the site does not have access to the video.

#### Proposed changelog entry for your changes:

* No changelog entry needed (I think?)

### CODE REVIEW

**Review by jeherve — COMMENTED** [review]
I would recommend placing this new endpoint with the rest of the wpcom v2 endpoints here: `_inc/lib/core-api/wpcom-endpoints/`

You'll also be able to take inspiration from the existing endpoints there.

**Comment by roundhill** [comment]
> I would recommend placing this new endpoint with the rest of the wpcom v2 endpoints here: `_inc/lib/core-api/wpcom-endpoints/`

Helpful! Thank you.

**Comment by roundhill** [comment]
WP.com side of this was deployed yesterday.

**Review by jgcaruso — DISMISSED** [review]
Works as described! Request succeeds when authenticated, and prevents unauthenticated requests.

Works well when testing with https://github.com/Automattic/jetpack/pull/17659 also

File: _inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-videopress.php
Comment (jeherve) [inline _inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-videopress.php:99]: Maybe we could expand a bit on this to provide more meaningful errors when something fails?

Something like this?

```suggestion
		if ( is_wp_error( $result ) ) {
			return rest_ensure_response( $result );
		}

		$response_body = json_decode( wp_remote_retrieve_body( $result ) );

		if (
			! empty( $response_body->code )
			&& 200 !== $response_body->code
		) {
			$error = new WP_Error(
				$response_body->code,
				$response_body->message,
				$response_body->data
			);

			return rest_ensure_response( $error );
		}

		return rest_ensure_response( $response_body );
```

File: _inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-videopress.php
Comment (jeherve) [inline _inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-videopress.php:58]: Maybe we can add `sanitize_callback` to all those arguments, to clean things up a bit before it gets sent to WordPress.com?

**Review by jeherve — CHANGES_REQUESTED** [review]

File: _inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-videopress.php
Comment (roundhill) [reply _inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-videopress.php]: Added something-ish like that, nice idea. @jgcaruso I also thought returning an object may make more sense than just a boolean, so you'll want to look for something like this for a successful response in your PR:

`{code: "success", message: "Video meta updated successfully.", data: 200}`

**Review by jeherve — APPROVED** [review]

**Review by jeherve — APPROVED** [review]
Re-approving after merging latest `master` to fix unrelated test failures.

**Comment by roundhill** [comment]
@jeherve do I need to worry about the `E2E Tests / E2E tests: with no plugin (pull_request)` test failing?

**Comment by jeherve** [comment]
> @jeherve do I need to worry about the `E2E Tests / E2E tests: with no plugin (pull_request)` test failing?

No, that's alright, it's not a blocking test.

**Comment by jeherve** [comment]
r218434-wpcom

**Pull request merged**
